examples/llm/src/models/layers/attention.py

```diff
@@ -256,6 +291,7 @@ def __init__(self, cfg: DictConfig, device: Optional[str] = None):
 
     def forward(self,
                 x,
+                past_key_value=None,
```

Comment:
Keep key_padding_mask as the second argument? As that's the most common one during training.

Reply:
x and past_key_value are activations so I kept them together.
key_padding_mask is a mask that defines what we do with the activations so I moved that after the activations.

Reply:
its also how [hf does it](https://github.com/huggingface/transformers/blob/c07a02a4b7892edfee22cbe57d3cdd9e10ae7a4d/src/transformers/models/gpt2/modeling_gpt2.py#L1050C11-L1065).
keeping past_key_value as second arg.

Reply:
They don't have key_padding_mask though. Anyway just a suggestion so what you have is fine.